### ISSUE TITLE
Is caffe-style pretrained models available?

### ISSUE THREAD

**Issue by wangg12** [open-mmlab/mmdetection#23]
Detectron.pytorch allows us to finetune from Detectron weights. Is this possible for this repo or do you provide the pretrained weights which are trained in caffe style? Somehow the performance of caffe style models is better.

**Comment by hellock** [comment]
We encourage to adopt pytorch-style backbones since it is more convenient to load pretrained weights from the model zoo with pytorch built-in apis.
In our experiments, caffe-style models generally perform slightly better in 1x lr schedule but fall behind in 2x schedule.

Here are some model weights
- [RPN](https://s3.ap-northeast-2.amazonaws.com/open-mmlab/mmdetection/models/rpn_r50_caffe_fpn_1x_20181010-8dc51276.pth)
- [Faster R-CNN](https://s3.ap-northeast-2.amazonaws.com/open-mmlab/mmdetection/models/faster_rcnn_r50_caffe_fpn_1x_20181010-37eb5008.pth)
- [Mask R-CNN](https://s3.ap-northeast-2.amazonaws.com/open-mmlab/mmdetection/models/mask_rcnn_r50_caffe_fpn_1x_20181010-2889f3ac.pth)
- [Fast R-CNN](https://s3.ap-northeast-2.amazonaws.com/open-mmlab/mmdetection/models/fast_rcnn_r50_caffe_fpn_1x_20181010-a9032e5e.pth)
- [Fast R-CNN (with mask)](https://s3.ap-northeast-2.amazonaws.com/open-mmlab/mmdetection/models/fast_mask_rcnn_r50_caffe_fpn_1x_20181010-032efee0.pth)


To use caffe-style backbones, you need to make some modifications in the config file.
- modify the path of pretrained weights
- modify `style='pytorch'` to `style='caffe'`
- add `bn_frozen=True` in backbone config.
- `img_norm_cfg = dict(mean=[102.9801, 115.9465, 122.7717], std=[1.0, 1.0, 1.0], to_rgb=False)`

**Comment by wangg12** [comment]
Thanks. @hellock

**Comment by bowenc0221** [comment]
Hi @hellock 

Did you use the original caffe ResNet models from [Kaiming's release](https://github.com/KaimingHe/deep-residual-networks)?  

I trained a caffe-style ResNet-50 with PyTorch, the pretrained model has Top-1 Accuracy: 75.53%. But when I used this pretrained model to initialize Faster RCNN FPN, I only got AP=35.8 with 1x schedule. Since it is a PyTorch pretrained model, I only modified `style='pytorch'` to `style='caffe'`. I tried both `bn_frozen=True` and `bn_frozen=False` and found no much difference.

**Comment by myownskyW7** [comment]
Hi @bowenc0221 

Our caffe ResNet model is converted from that of Detectron.  If you set **bn_eval=True** and **bn_frozen=True**, there should not be any problem. You may have another try.

**Comment by thangvubk** [comment]
@myownskyW7
Did you benchmark top 1 acc on imagenet after converting caffe2 resnet.

**Comment by lzy00codeforfun** [comment]
> Hi @bowenc0221
> Our caffe ResNet model is converted from that of Detectron.  You may set bn_eval=True and bn_frozen=True and you should also set img_norm_cfg = dict(
> mean=[102.9801, 115.9465, 122.7717], std=[1.0, 1.0, 1.0], to_rgb=False).
Hi, I recently just convert the caffe model from detectron `'.pkl'` to pytorch model `'.pth'. However, when I use this `'.pth'`
` to train and modify the config **img_norm_cfg = dict(
mean=[102.9801, 115.9465, 122.7717], std=[1.0, 1.0, 1.0], to_rgb=False).**、 **type='pytorch'**,the result doesn't seem equal to the origin pytorch. Could I know how you convert the Detectron model?

**Comment by lzy00codeforfun** [comment]
@myownskyW7 
Hi, I recently just convert the caffe model from detectron` '.pkl'` to pytorch model `'.pth'`. However, when I use this `'.pth'` to train and modify the config **img_norm_cfg = dict(
mean=[102.9801, 115.9465, 122.7717], std=[1.0, 1.0, 1.0], to_rgb=False).**、 **type='pytorch'**,the result doesn't seem equal to the origin pytorch. Could I know how you convert the Detectron model?

**Comment by jinxuan777** [comment]
@myownskyW7 can you share your code for converting model from detectron ?

**Comment by myownskyW7** [comment]
@lzy00codeforfun @jinxuan777 Please see https://github.com/open-mmlab/mmdetection/pull/709
